### PR TITLE
Demonstration of migration to single build.sbt config

### PR DESCRIPTION
**Pull request by travisbrown**
This is a quick demonstration that combines all the SBT build files into a single build.sbt file, removes some usage of deprecated methods (e.g. `Defaults.defaultSettings`), etc. It should do exactly the same things as the current build, but I haven't checked it in detail.

I personally find this format a little nicer and more maintainable, but I'm not set on advocating for it—I just wanted to show how it would work.

### CODE REVIEW

**Comment by non** [comment]
This looks great! I think this will be a lot more maintainable.

:+1:

**Comment by mpilquist** [comment]
:+1:

File: build.sbt [hunk 1]
```diff
@@ -1,64 +1,155 @@
...
+  "org.scalacheck" %% "scalacheck" % "1.11.3",
+  "org.typelevel" %% "discipline" % "0.2.1"
 )
 
-resolvers in ThisBuild ++=
-  ("bintray/non" at "http://dl.bintray.com/non/maven") ::
-  Resolver.sonatypeRepo("releases") ::
-  Nil
+lazy val aggregate = project.in(file("."))
+  .settings(catsSettings: _*)
```
Comment (dwijnand) [inline build.sbt]: looks like this is missing `noPublishSettings`

Comment (travisbrown) [inline build.sbt]: Ah, good catch @dwijnand—want to submit a pull request or should I?

Comment (dwijnand) [inline build.sbt]: https://github.com/non/cats/pull/38

**Pull request merged**